### DISCUSSION TITLE
Song select filter templates or filter cheatsheet

### DISCUSSION THREAD

**Discussion by Ladwick** [ppy/osu#31369] [Ideas] [answered]
This is an idea I had while using filters many times, and even finding new ones recently by visiting the wiki, actually we have a few filters for searching and we will probably get more filters in the future. To my knowloedge, there is no way of knowing these aside from asking someone or searching in the wiki or the source code, things that the average user probably won't do, so I thought about two possible solutions to this.

### Filter templates
I know there is a song select redesign in the works so maybe this should wait for whenever that happens, but a few buttons that introduce the filter formula would be nice for saving time but mainly to allow the casual user to know that there are filters for searching beatmaps with specific parameters in the first place. 

As for the implementation, maybe a button next to the search icon that displays a dropdown on click (or hover?) with all the filters and operators available (maybe even suggestions for the working values of filters like "status", though the point of the idea is to allow the user to know there are filters in the first place, from that point on, they will be able to figure the rest themselves I think.)
![image](https://github.com/user-attachments/assets/00dd8d3f-f039-4b3e-b24c-e9cd724121a4)
![image](https://github.com/user-attachments/assets/9db69d8c-d8ac-4301-835e-486b5886ff60)

### Filter cheatsheet
This probably would be a easier implementation, having all the filters somewhere in the screen or behind an option in the footer, something like options>filters that would display a cheatsheet with every filter, operator and specific value there is.

**Comment by peppy** [comment, marked as answer]
I'd probably add this as a tooltip to the text box.
